src/helpers.js

```diff
@@ -222,26 +222,58 @@ export default class Helpers {
     return el
   }
 
-  sanitize(html) {
-    if (typeof html === 'string' && html.trim() === '')
-      return ''
-    else
-      return DOMPurify.sanitize(html, { ADD_TAGS: ['use'] })
+  sanitize(data) {
+    if (data === null || data === undefined) {
+      return data
+    }
+
+    if (typeof data === 'string' && data.trim() !== '') {
+      let sanitized = DOMPurify.sanitize(data, { ADD_TAGS: ['use'] })
+
+      // Additional sanitization for patterns that DOMPurify doesn't catch in standalone strings
+      if (/^\s*(javascript|vbscript|data\s*:\s*text\/html)/i.test(sanitized)) {
+        return ''
+      }
+
+      return sanitized
+    }
+
+    if (Array.isArray(data)) {
+      return data.map(item => this.sanitize(item))
+    }
+
+    if (typeof data === 'object') {
+      const sanitized = {}
+      for (const [key, value] of Object.entries(data)) {
+        sanitized[key] = this.sanitize(value)
+      }
+      return sanitized
+    }
+
+    // For primitives (numbers, booleans, etc.), return as-is
+    return data
   }
 
   // Templates
-  render(template, data) {
+  render(template, data, options = {}) {
+    const defaults = { sanitize: true }
+    options = Object.assign({}, defaults, options)
+
+    // Sanitize HTML for XSS protection
+    if (options.sanitize) {
+      data = sanitize(data)
+    }
+
     return App.templates.render(template, data)
   }
 
-  insertTemplate(query, template, data, position = null) {
-    const html = render(template, data)
+  insertTemplate(query, template, data, options = {}) {
+    const defaults = { sanitize: true }
```

Comment:
The original idea of this method is to use with "sanitize false", as we only need to sanitize data, the template is controlled by us. Otherwise I'm not sure if this method makes sense 🤔?

In general, I'd like to keep Ralix rendering as safe as possible by default, but at the same time we need to provide good ergonomics to work with the framework (passing "sanitize false" seems like an exception).